### PR TITLE
Adding xc loss

### PR DESCRIPTION
**Pull request by advikavs**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/losses.py [hunk 1]
```diff
@@ -1263,7 +1263,7 @@ def loss(substruct_rep, overlapped_node_rep, context_rep,
         return loss
 
 
-class DensityProfileLoss(Loss):
+class XCLoss(Loss):
```
Comment (rbharath) [inline deepchem/models/losses.py:1266]: Quick reminder to update the name in the docs if you would like to change the name.

File: deepchem/models/losses.py [hunk 2]
```diff
@@ -1294,19 +1294,16 @@ class DensityProfileLoss(Loss):
...
-        """
+    def _create_pytorch_loss(self):
+
         import torch
 
         def loss(output, labels):
             output, labels = _make_pytorch_shapes_consistent(output, labels)
-            return torch.sum((labels - output)**2 * volume)
-
+            if output.shape[0] > 1 and len(output.shape) == 1:
```
Comment (rbharath) [inline deepchem/models/losses.py:1303]: Can you add a description of the calculation to the docstring?

File: deepchem/models/dft/dftxc.py [hunk 3]
```diff
@@ -74,6 +74,11 @@ def forward(self, inputs):
                 qcs.append(evl.run(system))
             if entry.entry_type == 'dm':
                 output.append((torch.as_tensor(entry.get_val(qcs)[0])))
+            elif entry.entry_type == 'dens':
```
Comment (rbharath) [inline deepchem/models/dft/dftxc.py:77]: Can you add a new unit test for this behavior?

File: deepchem/models/losses.py [hunk 4]
```diff
@@ -1294,19 +1294,16 @@ class DensityProfileLoss(Loss):
...
+                return torch.sum(output)
+            else:
+                return( L2Loss()._create_pytorch_loss()(output, labels))
```
Comment (rbharath) [inline deepchem/models/losses.py:1306]: Can you add an updated unit test for this functionality?

**Review by rbharath — COMMENTED** [review]
Changes look fine, but we will need unit tests and docs

File: deepchem/models/dft/dftxc.py (hunk 3 above)
Comment (advikavs) [reply deepchem/models/dft/dftxc.py:77]: Added

File: deepchem/models/losses.py (hunk 4 above)
Comment (advikavs) [reply deepchem/models/losses.py:1306]: Added

Comment (advikavs) [reply deepchem/models/losses.py:1306]: Added

File: deepchem/models/losses.py (hunk 2 above)
Comment (advikavs) [reply deepchem/models/losses.py:1305]: Done

File: deepchem/models/dft/dftxc.py [hunk 5]
```diff
@@ -74,6 +74,13 @@ def forward(self, inputs):
...
```
Comment (rbharath) [inline deepchem/models/dft/dftxc.py:77]: We should document this calculation in the docstring. We also need to explain we don't support predictions for density profiles at present

Comment (rbharath) [reply deepchem/models/dft/dftxc.py:77]: We should provide the needed outputs for the loss computation but perform the actual loss in the XCLoss. Have outputs
```
val, volume
```
from the forward pass for density formulas

File: deepchem/models/losses.py [hunk 6]
```diff
@@ -1294,18 +1296,16 @@ class DensityProfileLoss(Loss):
...
-            Shape of the tensor depends on the molecule/crystal and the integration grid
-        """
+    def _create_pytorch_loss(self):
+
         import torch
 
         def loss(output, labels):
             output, labels = _make_pytorch_shapes_consistent(output, labels)
-            return torch.sum((labels - output)**2 * volume)
+            if output.shape[0] > 1 and len(output.shape) == 1:
```
Comment (rbharath) [inline deepchem/models/losses.py:1305]: Instead of this, have
```
true_val - val)**2 * volume
```
